sktime/annotation/tests/test_plotting.py

```diff
@@ -0,0 +1,75 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+from sktime.annotation.plotting.utils import (
+    plot_time_series_with_change_points,
+    plot_time_series_with_profiles,
+)
+from sktime.utils.validation._dependencies import _check_soft_dependencies
+
+
+@pytest.fixture
+def time_series_data():
+    ts_data = np.random.rand(100)
+    ts = pd.DataFrame({"Data": ts_data})
+    true_cps = [4, 8]
+    font_size = 12
+    ts_name = "Test Time Series"
+    profiles = np.array([np.random.rand(100) for _ in range(20)])
+    found_cps = [35, 65]
+    score_name = "Custom Score"
+    return {
+        "ts_name": ts_name,
+        "ts": ts,
+        "true_cps": true_cps,
+        "font_size": font_size,
+        "profiles": profiles,
+        "found_cps": found_cps,
+        "score_name": score_name,
+    }
+
+
+@pytest.mark.skipif(
+    not _check_soft_dependencies("matplotlib", severity="none"),
+    reason="Seaborn is required as a dependency for this plot.",
+)
+def test_plot_time_series_with_change_points(time_series_data):
+    import matplotlib.pyplot as plt
+
+    # Access data from the fixture
+    ts_name = time_series_data["ts_name"]
+    ts = time_series_data["ts"]
+    true_cps = time_series_data["true_cps"]
+    font_size = time_series_data["font_size"]
+
+    fig, ax = plot_time_series_with_change_points(ts_name, ts, true_cps, font_size)
+
+    assert isinstance(fig, plt.Figure)
+    assert isinstance(ax, plt.Axes)
+    assert ax.get_title() == ts_name
+
+
+@pytest.mark.skipif(
+    not _check_soft_dependencies("seaborn", "matplotlib", severity="none"),
```

Comment:
Non-blocking comment: since `matplotlib` is a dependency of `seaborn`, checking for `seaborn` is probably sufficient.

Reply:
well, I'd say, better be sure. We had it in the past where we relied on X being dependency of Y and tested only Y which worked well until Y dropped X. In the painful example, Y=`pytest` and X=`attrs`.

That taught me, that all direct import locations should also be `pyproject` dependencies, and when testing we should also test for direct dependencies.